pkg/k8s/namespaces/crud.go

```diff
@@ -21,3 +27,100 @@ func IsOktetoNamespace(ns string, c *kubernetes.Clientset) bool {
 	}
 	return n.Labels[OktetoLabel] == "true"
 }
+
+//CheckAvailableResources checks if therre are available resources for activating the dev env
+func CheckAvailableResources(dev *model.Dev, create bool, d *appsv1.Deployment, c *kubernetes.Clientset) error {
+	needCPU := resource.MustParse("0")
+	needMemory := resource.MustParse("0")
+	if !existsSyncthing(dev, c) {
+		needCPU.Add(resource.MustParse("500m"))
+		needMemory.Add(resource.MustParse("256Mi"))
+	}
+	lr, err := c.CoreV1().LimitRanges(dev.Namespace).Get(dev.Namespace, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	needCPU.Add(matchCPU(dev, true, 1, d.Spec.Template.Spec, lr))
+	needMemory.Add(matchMemory(dev, true, 1, d.Spec.Template.Spec, lr))
+	if !create {
+		needCPU.Sub(matchCPU(dev, false, d.Status.Replicas, d.Spec.Template.Spec, lr))
+		needMemory.Sub(matchMemory(dev, false, d.Status.Replicas, d.Spec.Template.Spec, lr))
+	}
+	rq, err := c.CoreV1().ResourceQuotas(dev.Namespace).Get(dev.Namespace, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	usedCPU := rq.Status.Used[apiv1.ResourceLimitsCPU]
+	usedCPU.Add(needCPU)
+	totalCPU := rq.Status.Hard[apiv1.ResourceLimitsCPU]
+	if usedCPU.Value() > totalCPU.Value() {
+		usedCPU.Sub(needCPU)
+		totalCPU.Sub(usedCPU)
+		return fmt.Errorf("Activating your dev environment requires %s cpu and your remaining quota is %s cpu", needCPU.String(), totalCPU.String())
+	}
+	usedMemory := rq.Status.Used[apiv1.ResourceLimitsMemory]
+	usedMemory.Add(needMemory)
+	totalMemory := rq.Status.Hard[apiv1.ResourceLimitsMemory]
+	if usedMemory.Value() > totalMemory.Value() {
+		usedMemory.Sub(needMemory)
+		totalMemory.Sub(usedMemory)
+		return fmt.Errorf("Activating your dev environment requires %s of memory and your remaining quota is %s of memory", needMemory.String(), totalMemory.String())
+	}
+	return nil
+}
+
+func existsSyncthing(dev *model.Dev, c *kubernetes.Clientset) bool {
+	ss, err := c.AppsV1().StatefulSets(dev.Namespace).Get(dev.GetStatefulSetName(), metav1.GetOptions{})
+	if err != nil || ss.Name == "" {
+		return false
+	}
+	return true
+}
+
+func matchCPU(dev *model.Dev, isDev bool, replicas int32, spec apiv1.PodSpec, lr *apiv1.LimitRange) resource.Quantity {
+	cpuPod := resource.MustParse("0")
+	for _, c := range spec.Containers {
+		v, ok := dev.Resources.Limits[apiv1.ResourceCPU]
+		if !isDev || !ok {
+			if c.Resources.Limits == nil {
+				cpuPod.Add(lr.Spec.Limits[0].Default[apiv1.ResourceCPU])
+			} else {
+				cpuPod.Add(c.Resources.Limits[apiv1.ResourceCPU])
+			}
+		} else {
+			cpuPod.Add(v)
+		}
+	}
+	if isDev {
+		return cpuPod
```

Comment:
is this branch because we only have a single replica of dev? Could we remove this branch and just let the loop in line 99 take care of it?